feat(WAF): waf support rule precise protection resource (#2984)
huaweicloud/terraform-provider-huaweicloud: docs/resources/waf_rule_precise_protection.md

---
subcategory: "Web Application Firewall (WAF)"
---

# huaweicloud_waf_rule_precise_protection

> Manages a WAF precise protection rule resource within HuaweiCloud.

Reviewer: ```suggestion
Manages a WAF rule precise protection resource within HuaweiCloud.
```

    **not_contain_all**, **equal_any**, **not_equal_all**, **equal_any**, **not_equal_all**, **prefix_any**,
    **not_prefix_all**, **suffix_any**, **not_suffix_all**, **len_greater**, **len_less**, **len_equal**,
    **len_not_equal**.

  + If `field` is set to **ip**: Valid values are **equal**, **not_equal**, **equal_any**, **not_equal_all**.

  + If `field` is set to **method**: Valid values are **equal**, **not_equal**.

  + If `field` is set to **request_line** or **request**: Valid values are **len_greater**, **len_less**, **len_equal**,
    **len_not_equal**.

  + If `field` is set to **params** or **cookie** or **header**: Valid values are **contain**, **not_contain**,
    **equal**, **not_equal**, **prefix**, **not_prefix**, **suffix**, **not_suffix**, **contain_any**,
    **not_contain_all**, **equal_any**, **not_equal_all**, **equal_any**, **not_equal_all**, **prefix_any**,
    **not_prefix_all**, **suffix_any**, **not_suffix_all**, **len_greater**, **len_less**, **len_equal**,
    **len_not_equal**, **num_greater**, **num_less**, **num_equal**, **num_not_equal**, **exist**, **not_exist**.

> * `subfield` - (Optional, String) Specifies the subfield of the condition. The parameter is required when field

Reviewer 2: ```suggestion
* `subfield` - (Optional, String) Specifies the subfield of the condition. The parameter is required when `field`
```

  is set to `params`, `header` and `cookie`.

  + If `field` is set to **cookie**: The parameter indicates cookie name.

  + If `field` is set to **params**: The parameter indicates param name.

  + If `field` is set to **header**: The parameter indicates an option in the header.

* `content` - (Optional, String) Specifies the content of the match condition. It is required when the `logic`
  does not end with `any` or `all`.

* `reference_table_id` - (Optional, String) Specifies the reference table id. It is required when the `logic`
  end with `any` or `all`. The type of reference table should be consistent with the type of field.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The ID of a precise protection rule.

## Import

There are two ways to import state, include `Without Enterprise Project ID` and `With Enterprise Project ID`.

### Without Enterprise Project ID

> The waf precise protection rule can be imported using `policy_id`, `rule_id`, separated by slashes, e.g.

Reviewer 2: ```suggestion
The waf precise protection rule can be imported using `policy_id`, `rule_id` separated by slashes, e.g.
```

> The waf precise protection rule can be imported using `policy_id`, `rule_id`, separated by slashes, e.g.

Reviewer: ```suggestion
The waf precise protection rule can be imported using `policy_id`, `rule_id`, separated by a slash, e.g.
```

```bash
$ terraform import huaweicloud_waf_rule_precise_protection.test <policy_id>/<rule_id>
```

### With Enterprise Project ID

> The waf precise protection rule can be imported using `policy_id`, `rule_id`, `enterprise_project_id`,

Reviewer 2: ```suggestion
The waf precise protection rule can be imported using `policy_id`, `rule_id`, `enterprise_project_id`
```

  Changing this parameter will create a new resource.

* `name` - (Required, String) Specifies the name of WAF precise protection rule.

* `priority` - (Required, Int) Specifies the priority of a rule. Smaller values correspond to higher priorities.
  If two rules are assigned with the same priority, the rule added earlier has higher priority.
  The value ranges from 0 to 1000.

* `conditions` - (Required, List) Specifies the match condition list.
  The [conditions](#RulePreciseProtection_conditions) structure is documented below.

* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID of WAF precise protection
  rule.

  Changing this parameter will create a new resource.

* `action` - (Optional, String) Specifies the protective action of WAF precise protection rule.
  Valid values are **block**, **pass**, **log**. The default value is **block**.

* `status` - (Optional, Int) Specifies the status of WAF precise protection rule.
  Valid values are as follows:
  + **0**: Disabled.
  + **1**: Enabled.

  The default value is **1**.

* `start_time` - (Optional, String) Specifies the time when the precise protection rule takes effect.

>   The time format is **yyyy-MM-dd HH:mm:ss**, e.g. **2006-01-02 15:04:05**.

Reviewer 3: it's better to use the recent time as an example.
